### ISSUE TITLE
Cannot run haproxy ingress with non root user

### ISSUE THREAD

**Issue by swapnil1988** [jcmoraisjr/haproxy-ingress#327]
Is it possible to run as a non-root user? 

Based on [this ](https://github.com/jcmoraisjr/haproxy-ingress/issues/292#issuecomment-457407924)comment I tried to change permissions/owner of /ingress-controller directory in a custom image but still I am not able to run haproxy ingress as a non-root user. 

**Custom image:**
```
FROM quay.io/jcmoraisjr/haproxy-ingress:v0.7.1

RUN addgroup -g 2000 group1      \
 && adduser -u 1000 -g group1 -D user1 \
 && adduser user1 root                        \
 && chown -R user1 /ingress-controller         \
 && chmod -R g+rwx /ingress-controller          \
 && chown user1 /haproxy-ingress-controller    \
 && chown user1 /haproxy-reload.sh             \
 && chown user1 /dumb-init                     \
 && chown user1 /docker-entrypoint.sh          \
 && chown user1 /start.sh                      \
 && chown -R user1 /tmp

ENV TMPDIR=/tmp

USER user1
```

**Throws below error:**

`F0421 14:01:52.638468       7 controller.go:1549] Error generating self signed certificate: could not create temp pem file /ingress-controller/ssl/default-fake-certificate.pem: open /ingress-controller/ssl/default-fake-certificate.pem799615147: permission denied`

**Comment by jcmoraisjr** [comment]
Hi, haproxy-ingress wasn't designed to run as non-root but I can confirm this is a recurrent feature. I'll give this a try and will let you know. v0.8+ will make a lot of stuff easier.

**Comment by swapnil1988** [comment]
Did manage to run with non root. Had to rebuild the docker image, gave permission/chown to required directories and changed http/https ports.

**Comment by ddanboyle** [comment]
Would you  post the docker image you used to run with non-root, so anybody else that might get here (like me) can build off your work?

**Comment by swapnil1988** [comment]
Here is the Dockerfile

`FROM quay.io/jcmoraisjr/haproxy-ingress@v0.7.1

RUN mkdir -p /etc/haproxy/modsecurity                      \
 && mkdir -p /etc/haproxy/template                         \
 && mkdir -p /ingress-controller/ssl                       \
 && mkdir -p /ingress-controller/cacerts                   \
 && mkdir -p /run/haproxy                                  \
 && mkdir -p /var/lib/haproxy                              \
 && touch  /etc/haproxy/haproxy.cfg                        \
 && chown -R 65534 /ingress-controller                     \
 && chgrp -R 65534 /ingress-controller                     \
 && chmod -R g+rwx /ingress-controller                     \
 && chown -R 65534 /haproxy-ingress-controller             \
 && chgrp -R 65534 /haproxy-ingress-controller             \
 && chown 65534 /haproxy-reload.sh                         \
 && chgrp 65534 /haproxy-reload.sh                         \
 && chown -R 65534 /dumb-init                              \
 && chgrp -R 65534 /dumb-init                              \
 && chown 65534 /docker-entrypoint.sh                      \
 && chgrp 65534 /docker-entrypoint.sh                      \
 && chown 65534 /start.sh                                  \
 && chgrp 65534 /start.sh                                  \
 && chown -R 65534 /home                                   \
 && chgrp -R 65534 /home                                   \
 && chown -R 65534 /etc                                    \
 && chgrp -R 65534 /etc                                    \
 && chown -R 65534 /tmp                                    \
 && chgrp -R 65534 /tmp                                    \
 && chown -R 65534 /run                                    \
 && chgrp -R 65534 /run                                    \
 && chown -R 65534 /var/run                                \
 && chown -R 65534 /var/run                                \
 && chgrp -R 65534 /var/lib                                \
 && chgrp -R 65534 /var/lib                                \
 && chmod -R ug+rwx /var/lib/haproxy                       \
 && apk add libselinux-utils

ENV TMPDIR=/tmp

USER 65534`

**Comment by jcmoraisjr** [comment]
Hi, implemented in v0.9-beta.1. See the [security](https://haproxy-ingress.github.io/docs/configuration/keys/#security) doc. Let me know if you have any problem. Closing.
